includes/csm-mode.php

```diff
@@ -0,0 +1,116 @@
+<?php
+// Exit if accessed directly
+if(!defined('ABSPATH')){
+	exit;
+}
+
+/**
+ * Enable Coming Soon/Maintenance Mode
+ */
+class Fiber_Admin_CSM_Mode{
+	private $mode = '';
+	
+	public function __construct(){
+		// Enable Coming Soon/Maintenance Mode
+		$this->mode = fiad_get_csm_mode_option('mode');
+		$this->fiad_create_template_if_not_exists();
+		add_action('template_redirect', [$this, 'fiad_preview_csm_page']);
+		if(fiad_get_csm_mode_option('enable')){
+			add_filter('template_include', [$this, 'fiad_csm_content']);
+			add_action('wp_head', [$this, 'fiad_csm_extra_css']);
+			add_action('wp_footer', [$this, 'fiad_csm_extra_js']);
+		}
+	}
+	
+	//No Header & Footer Page
+	public function fiad_csm_content($template){
+		if(!current_user_can('edit_themes') || !is_user_logged_in()){
+			$new_template = dirname(__FILE__) . '/templates/' . $this->mode . '.php';
```

Comment:
- Check and reuse Fiber Admin constants instead of repeating `dirname(__FILE__)`
- Define another constant / variable if needed for the all duplicate `dirname(__FILE__) . '/templates/'` that you used below